### PR TITLE
fix: add StateLibError type in docs

### LINKED ISSUE #79: StateLibError not defined error when using lib in typescript
### Expected Behaviour
using lib in typescript application it will build without error

### Actual Behaviour
Action Cleaning Done
node_modules/@adobe/aio-lib-state/types.d.ts:79:25 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

79     ERROR_BAD_ARGUMENT: StateLibError;
                           ~~~~~~~~~~~~~

node_modules/@adobe/aio-lib-state/types.d.ts:80:24 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

80     ERROR_BAD_REQUEST: StateLibError;
                          ~~~~~~~~~~~~~

node_modules/@adobe/aio-lib-state/types.d.ts:81:28 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

81     ERROR_NOT_IMPLEMENTED: StateLibError;
                              ~~~~~~~~~~~~~

node_modules/@adobe/aio-lib-state/types.d.ts:82:30 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

82     ERROR_PAYLOAD_TOO_LARGE: StateLibError;
                                ~~~~~~~~~~~~~

node_modules/@adobe/aio-lib-state/types.d.ts:83:28 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

83     ERROR_BAD_CREDENTIALS: StateLibError;
                              ~~~~~~~~~~~~~

node_modules/@adobe/aio-lib-state/types.d.ts:84:21 - error TS2552: Cannot find name 'StateLibError'. Did you mean 'StateLibErrors'?

84     ERROR_INTERNAL: StateLibError;
               
Build fails

### Reproduce Scenario (including but not limited to)

#### Steps to Reproduce
typescript app and try to use the state lib in a function.  
 "typescript": "^4.6.4"
For the record this worked error free over a year ago. 

#### Platform and Version
System:
    OS: Windows 10 10.0.22000
    CPU: (20) x64 Intel(R) Xeon(R) W-2255 CPU @ 3.70GHz
    Memory: 90.87 GB / 127.69 GB
  Binaries:
    Node: 14.19.1
    Yarn: 1.22.18
    npm: 6.14.16
  Virtualization:
    Docker: Not Found
  npmGlobalPackages:
    @adobe/aio-cli: Not Found

  Proxies:
    http: (not set)
    https: (not set)
  CLI plugins:
    core:
      @adobe/aio-cli 8.3.0
      @adobe/aio-cli-plugin-app 8.6.1
      @adobe/aio-cli-plugin-auth 2.6.0
      @adobe/aio-cli-plugin-certificate 0.3.1
      @adobe/aio-cli-plugin-config 3.0.1
      @adobe/aio-cli-plugin-console 3.4.2
      @adobe/aio-cli-plugin-events 1.1.6
      @adobe/aio-cli-plugin-info 2.1.0
      @adobe/aio-cli-plugin-runtime 5.4.0
      @oclif/plugin-autocomplete 0.3.0
      @oclif/plugin-help 3.3.1
      @oclif/plugin-not-found 2.3.1
      @oclif/plugin-plugins 1.10.11
      @oclif/plugin-warn-if-update-available 1.7.3
    user:
    link:

#### Sample Code that illustrates the problem
tsconfig.json
`{
    "compilerOptions": {
        /* Basic Options */
        "module": "commonjs",  /*Specify module code generation: "None", "CommonJS", "AMD", "System", "UMD", "ES6", "ES2015" or "ESNext".  Only "AMD" and "System" can be used in conjunction with --outFile. "ES6" and "ES2015" values may be used when targeting "ES5" or lower.*/
        "target": "es5",  /*Specify ECMAScript target version: "ES3" (default),"ES5","ES6"/"ES2015","ES2016","ES2017","ES2018","ES2019","ES2020","ESNext"*/
        //"lib": [ "es2015", "dom" ],  /*List of library files to be included in the compilation. see https://www.typescriptlang.org/docs/handbook/compiler-options.html */
        //"removeComments": true,
        //"resolveJsonModule": true,
        //"esModuleInterop": true,
        //"typeRoots" : [
        //   "./node_modules/@types/",
        //    "./typings/"
        //],
        // "lib": [],                             /* Specify library files to be included in the compilation. */
        "allowJs": false,                       /* Allow javascript files to be compiled. */
        "checkJs": false,                       /* Report errors in .js files. */
        // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
        "declaration": false,                   /* Generates corresponding '.d.ts' file. */
        // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
        "sourceMap": false,                     /* Generates corresponding '.map' file. */
        // "outFile": "./",                       /* Concatenate and emit output to single file. */
        "outDir": "actions",                        /* Redirect output structure to the directory. */
        // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
        // "composite": true,                     /* Enable project compilation */
        // "removeComments": true,                /* Do not emit comments to output. */
        // "noEmit": true,                        /* Do not emit outputs. */
        //"importHelpers": true,                 /* Import emit helpers from 'tslib'. */
        // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
        // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */

        /* Strict Type-Checking Options */
        //"strict": true,                           /* Enable all strict type-checking options. */
        "noImplicitAny": false,                 /* Raise error on expressions and declarations with an implied 'any' type. */
        //"strictNullChecks": false,              /* Enable strict null checks. */
        // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
        // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
        // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
        // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
        
        /* Additional Checks */
        // "noUnusedLocals": true,                /* Report errors on unused locals. */
        // "noUnusedParameters": true,            /* Report errors on unused parameters. */
        // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
        // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
        
        /* Module Resolution Options */
        "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
        // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
        // "paths": {},                           /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
        // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
        // "typeRoots": [],                       /* List of folders to include type definitions from. */
        // "types": ["jest", "node"],                           /* Type declaration files to be included in compilation. */
        // "allowSyntheticDefaultImports": true,  /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
        "esModuleInterop": true                   /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
        // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
        
        /* Source Map Options */
        // "sourceRoot": "",                      /* Specify the location where debugger should locate TypeScript files instead of source locations. */
        // "mapRoot": "",                         /* Specify the location where debugger should locate map files instead of generated locations. */
        // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
        // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
        
        /* Experimental Options */
        // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
        // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
    },
    "include": [
        "./actions-src/**/*"
    ],
    "exclude": [
        "dev-keys"
    ]
}`

Is this a problem or do I not fully understand some type of definition rule or do i have a TypeScript check on I dont need.
![image (1)](https://user-images.githubusercontent.com/72889866/168390324-ac0e9c0f-8aae-4b48-8a9b-773b2acb7a4e.png)

### PR DESCRIPTION
**Pull request by shazron**
Closes #79 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: lib/StateStoreError.js [hunk 1]
```diff
@@ -14,10 +14,13 @@ const { ErrorWrapper, createUpdater } = require('@adobe/aio-lib-core-errors').Ai
 const logger = require('@adobe/aio-lib-core-logging')('@adobe/aio-lib-state', { provider: 'debug' })
 
 /**
- * @typedef StateLibError
- * @private
- *
+ * @typedef {Object} StateLibError
```
Comment (moritzraho) [inline lib/StateStoreError.js:17]: `Object` should be `object` according to linter errors below

File: lib/StateStoreError.js [hunk 2]
```diff
@@ -14,10 +14,13 @@ const { ErrorWrapper, createUpdater } = require('@adobe/aio-lib-core-errors').Ai
...
+ * @property {string} message The message for the Error
+ * @property {string} code The code for the Error
+ * @property {string} sdk The SDK associated with the Error
+ * @property {Object} sdkDetails The SDK details associated with the Error
```
Comment (moritzraho) [inline lib/StateStoreError.js:21]: same comment here

File: lib/StateStoreError.js (hunk 1 above)
Comment (shazron) [reply lib/StateStoreError.js]: fixed

File: lib/StateStoreError.js (hunk 2 above)
Comment (shazron) [reply lib/StateStoreError.js]: fixed

**Pull request merged**